### PR TITLE
Reason for payment is now list of fake reasons instead of random string ID to obfuscate fact its from the Bisq

### PR DESCRIPTION
**Pull request by ghost**
Solves https://github.com/bisq-network/bisq/issues/2869

image

----

For now, the list of reasons is very limited, and I need to find some more (like hundreds more) reasons. But from the code perspective it's working.

I did some research in the https://github.com/bisq-network/bisq/issues/2869 issue, and I doubt that Venmo is usable.

### CODE REVIEW

File: common/src/main/java/bisq/common/util/ReasonsForPayment.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package bisq.common.util;
+
+public class ReasonsForPayment {
+
+    private static final String[] REASONS = {
+            "Miss you",
```
Comment (ghost) [inline common/src/main/java/bisq/common/util/ReasonsForPayment.java:6]: Should we translate this via `resources/i18n`?

**Comment by cd2357** [comment]
NACK.

Interesting concept, but I think it adds complexity and might actually make it easier to match such a payment description to Bisq.

The random string used now is used a lot in other places as well (some merchants use it as Order ID, or Purchase ID, to match it with orders in their own payment systems).

Also, it is sometimes a random alpha-numeric string, sometimes only numeric. Which obfuscates it even more.

So I'd say it's good the way it is now.

**Comment by ghost** [comment]
True, I agree with your points. The only thing I like to solve is how to have issues ready for implementation, that were **accepted** and are actually **wanted** before anyone picks them up and spend time on them. 

I'll open this on Keybase chat as well.

**Comment by wiz** [comment]
NACK, mediators need the trade ID in the payment details field to resolve disputes

**Comment by dmos62** [comment]
NACK, small dictionary size means that this is private by obscurity (that's bad). If we wanted to increase dictionary size, we would have to use some kind of dynamic generation, but then how to make it resilient to heuristics? Plus what @wiz said about needing trade ID in the payments field. @petrhejna sorry for the bad experience. You happened upon an issue whose solution is easy to implement (most likely), but hard to find. @sqrrm I think there's a concensus that this PR should be closed.

**Comment by sqrrm** [comment]
Closing as not accepted.

**Pull request closed without merging**